### PR TITLE
[STTNHUB-216] Agenda: Show group headers for hidden items

### PR DESCRIPTION
**Pull request by MarkLark86**
## Checklist

- [x] This pull request is not adding new forms that use redux
- [x] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence
- [x] This pull request is replacing `lodash.get` with optional chaining for modified code segments

### CODE REVIEW

File: newsroom/agenda/agenda.py [hunk 1]
```diff
@@ -334,42 +353,12 @@ def _set_event_date_range(search):
...
-            {
-                "bool": {
-                    "filter": [
-                        {"term": {"dates.all_day": True}},
-                        {"range": {dates_field: {"gte": search.args["date_from"]}}},
-                    ],
-                },
-            },
-        ]
+        dates_field = "dates.start" if app.config.get("AGENDA_SHOW_MULTIDAY_ON_START_ONLY") else "dates.end"
```
Comment (petrjasek) [inline newsroom/agenda/agenda.py:357]: I think we wouldn't need this config anymore if we show those hidden items. that's like doing both at the same time.

**Review by petrjasek — APPROVED** [review]
looks ok to me, was wondering why not fetch all the items with single query and only hide some in the UI?

File: assets/interfaces/configs.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    show_list_action_icons?: {
+        large?: boolean;
+        compact?: boolean;
+        mobile?: boolean;
+    };
+    abstract?: {displayed?: boolean};
+    wordcount?: {displayed?: boolean};
+    charcount?: {displayed?: boolean};
+    slugline?: {displayed?: boolean};
+    contacts?: {displayed?: boolean};
```
[the remark this answers is not in the document]
Comment (MarkLark86) [reply assets/interfaces/configs.ts:39]: Good point, I wasn't sure if we assume it's there or not. I suppose we can

**Comment by MarkLark86** [comment]
@petrjasek I have made the suggested changes, and reverted back to getting the items in 1 request.

**Review by petrjasek — APPROVED** [review]

**Pull request merged**